crates/lakekeeper/src/service/tasks/mod.rs

```diff
@@ -277,6 +352,102 @@ pub struct Task {
     pub(crate) data: serde_json::Value,
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct TaskInfo {
+    pub task_metadata: TaskMetadata,
+    pub queue_name: TaskQueueName,
+    pub id: TaskAttemptId,
+    pub status: Option<TaskStatus>,
+    pub outcome: Option<TaskOutcome>,
```

Comment:
This introduces unneccessary ambiguity. It allows for a struct with no field set or both. Semantically, they are mutually exclusive and required, so we should encode this in a type.